Vertical.HubSpot.Api/Engagements/EngagementsApi.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Vertical.HubSpot.Api.Data;
+using Vertical.HubSpot.Api.Extensions;
+using Vertical.HubSpot.Api.Models;
+
+namespace Vertical.HubSpot.Api.Engagements
+{
+    /// <summary>
+    /// access to contacts api
+    /// </summary>
+    public class EngagementsApi : IEngagementsApi
+    {
+        private readonly HubSpotOptions _options;
+        readonly ModelRegistry models;
+        readonly HubSpotRestClient rest;
+
+        /// <summary>
+        /// creates a new <see cref="EngagementApi"/>
+        /// </summary>
+        /// <param name="rest">rest client to call api</param>
+        /// <param name="models">model registry used to access entity models</param>
+        internal EngagementsApi(HubSpotOptions options, HubSpotRestClient rest, ModelRegistry models) {
+            _options = options;
+            this.rest = rest;
+            this.models = models;
+        }
+
+        /// <summary>
+        /// get a contact by id
+        /// </summary>
+        /// <typeparam name="T">type of contact model</typeparam>
+        /// <param name="id">id of contact to return</param>
+        /// <returns>contact data</returns>
+        public async Task<T> GetEngagement<T>(long id)
+            where T : HubSpotEngagementResult
+        {
+            JObject response = await rest.Get<JObject>($"/engagements/v1/engagements/{id}");
+            return ToEngagementResult<T>(response);
+        }
+
+        public async Task<JObject> DeleteEngagement(long id)
+        {
+            JObject response = await rest.Delete<JObject>($"/engagements/v1/engagements/{id}");
```

Comment:
if you return the rest response directly here anyways you could drop the await/async statement and save a task wrap operation.

Reply:
Alright